### PR TITLE
feat: add ability to retrieve/update waitlist status for a user

### PR DESCRIPTION
**Pull request by brandonroberts**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR:

- Adds the ability to select and update the waitlist status for an individual user
- Adds the `is_waitlisted` flag to the users migration script

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -74,4 +77,26 @@ export class AuthController {
...
+  @Post("/waitlist")
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiOperation({
+    operationId: "postWaitlist",
+    summary: "Updates waitlist information for user",
+  })
+  @ApiOkResponse({ type: SupabaseAuthDto })
+  async postWaitlist (
+    @User() user: SupabaseAuthUser,
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:90]: you can use `@UserId` to get rid of the hard typing

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -74,4 +77,26 @@ export class AuthController {
...
+  ): Promise<SupabaseAuthDto> {
+    const { user_metadata: { sub: id } } = user;
+
+    try {
+      await this.userService.updateWaitlistStatus(id as number);
+    } catch (e) {
+      // handle error
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:97]: if you don't throw a better error don't try/catch it, nestjs exception filtering will still work

File: src/user/user.service.ts [hunk 3]
```diff
@@ -63,7 +63,17 @@ export class UserService {
     try {
       await this.findOneById(id);
 
-      await this.userRepository.update(id, { is_onboarded: true });
+      await this.userRepository.update(id, { is_onboarded: true, is_waitlisted: false });
+    } catch (e) {
+      // handle error
```
Comment (0-vortex) [inline src/user/user.service.ts:68]: same as the above try/catch comment

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Minor changes to keep the backend consistent 🍕

File: src/user/user.service.ts [hunk 4]
```diff
@@ -63,9 +63,19 @@ export class UserService {
...
+    }
+  }
+
+  async updateWaitlistStatus (id: number) {
+    try {
+      await this.findOneById(id);
+
+      await this.userRepository.update(id, { is_waitlisted: true });
+    } catch (e) {
+      throw new Error("Unable to update user waitlist status");
```
Comment (0-vortex) [inline src/user/user.service.ts:78]: If any of these errors make sense: https://docs.nestjs.com/exception-filters#built-in-http-exceptions try and update from normal error and add it to the return decorator on the api routes

**Review by 0-vortex — CHANGES_REQUESTED** [review]
1 more quick thing 🍕

File: src/user/user.service.ts (hunk 4 above)
Comment (brandonroberts) [reply src/user/user.service.ts]: Dome

**Review by 0-vortex — APPROVED** [review]
LGTM! 👍

**Pull request merged**